nodefs/syscall_linux.go

```diff
@@ -0,0 +1,20 @@
+// Copyright 2016 the Go-FUSE Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package nodefs
+
+import (
+	"syscall"
+	"unsafe"
+)
+
+// futimens - futimens(3) calls utimensat(2) with "pathname" set to null and
+// "flags" set to zero
+func futimens(fd int, times *[2]syscall.Timespec) (err error) {
+	_, _, e1 := syscall.Syscall6(syscall.SYS_UTIMENSAT, uintptr(fd), 0, uintptr(unsafe.Pointer(times)), uintptr(0), 0, 0)
+	if e1 != 0 {
+		err = syscall.Errno(e1)
+	}
+	return
+}
```

Comment:
I gocryptfs, I now do this:
```go
// FutimesNano syscall.
func FutimesNano(fd int, a *time.Time, m *time.Time) (err error) {
	ts := timesToTimespec(a, m)
	// To avoid introducing a separate syscall wrapper for futimens()
	// (as done in go-fuse, for example), we instead use the /proc/self/fd trick.
	procPath := fmt.Sprintf("/proc/self/fd/%d", fd)
	return unix.UtimesNanoAt(unix.AT_FDCWD, procPath, ts, 0)
}
```
Code: https://github.com/rfjakob/gocryptfs/blob/173218895c55f381750408d9452d118b8c574f58/internal/syscallcompat/sys_linux.go#L203

Tests: https://github.com/rfjakob/gocryptfs/blob/173218895c55f381750408d9452d118b8c574f58/tests/matrix/matrix_test.go#L675

Reply:
what is the problem with a syscall wrapper?

Reply:
No specific problem. My solution is not even less ugly, ignore the comment.